python-frontend/src/main/java/org/sonar/python/semantic/v2/SymbolsModuleTypeProvider.java

```diff
@@ -137,15 +136,21 @@ private PythonType convertToFunctionType(FunctionSymbol symbol, Map<Symbol, Pyth
     return functionType;
   }
 
-  private static ParameterV2 convertParameter(FunctionSymbol.Parameter parameter) {
-    return new ParameterV2(parameter.name(),
-      PythonType.UNKNOWN,
-      parameter.hasDefaultValue(),
-      parameter.isKeywordOnly(),
-      parameter.isPositionalOnly(),
-      parameter.isKeywordVariadic(),
-      parameter.isPositionalVariadic(),
-      null);
+  private PythonType resolveReturnType(Map<Symbol, PythonType> createdTypesBySymbol, @Nullable ClassSymbol classSymbol) {
+    var returnType = PythonType.UNKNOWN;
+    if (classSymbol == null) {
+      return returnType;
+    }
+    if (rootModule == null) {
+      return convertToType(classSymbol, createdTypesBySymbol);
+    }
+    // If the symbol is a built-in symbol and the root module already exists, we search for it instead of recreating the symbol
+    // This is necessary to avoid duplicated symbols in the original ProjectSymbolTable to translate into duplicated PythonType
+    returnType = Optional.ofNullable(classSymbol.fullyQualifiedName())
```

Comment:
it makes sense to leave a TODO comment here that this logic should be cleaned up once lazy type will be implemented